example/thirdparty/arrow/build.mill

```diff
@@ -0,0 +1,689 @@
+  val kotlinxSerializationCoreJs =
+    ivy"org.jetbrains.kotlinx:kotlinx-serialization-core-js:${versions.kotlinxSerialization}"
+  val kotlinxSerializationJson =
+    ivy"org.jetbrains.kotlinx:kotlinx-serialization-json:${versions.kotlinxSerialization}"
+  val kotlinxSerializationJsonJs =
+    ivy"org.jetbrains.kotlinx:kotlinx-serialization-json-js:${versions.kotlinxSerialization}"
+  val squareupOkhttpMockWebServer =
+    ivy"com.squareup.okhttp3:mockwebserver:${versions.mockWebServer}"
+  val squareupRetrofitLib = ivy"com.squareup.retrofit2:retrofit:${versions.retrofit}"
+  val squareupRetrofitConverterGson =
+    ivy"com.squareup.retrofit2:converter-gson:${versions.retrofit}"
+  val squareupRetrofitConverterMoshi =
+    ivy"com.squareup.retrofit2:converter-moshi:${versions.retrofit}"
+  val squareupRetrofitConverterKotlinxSerialization =
+    ivy"com.squareup.retrofit2:converter-kotlinx-serialization:${versions.retrofit}"
+  val squareupMoshiKotlin = ivy"com.squareup.moshi:moshi-kotlin:${versions.moshi}"
+  val ksp = ivy"com.google.devtools.ksp:symbol-processing-api:${versions.kspVersion}"
+  val kspGradlePlugin =
+    ivy"com.google.devtools.ksp:com.google.devtools.ksp.gradle.plugin:${versions.kspVersion}"
+  val classgraph = ivy"io.github.classgraph:classgraph:${versions.classgraph}"
+  val composeRuntime = ivy"androidx.compose.runtime:runtime:${versions.compose}"
+  val kotlinCompileTesting = ivy"dev.zacsweers.kctfork:core:${versions.kotlinCompileTesting}"
+  val kotlinCompileTestingKsp = ivy"dev.zacsweers.kctfork:ksp:${versions.kotlinCompileTesting}"
+  val cache4kJs = ivy"io.github.reactivecircus.cache4k:cache4k-js:${versions.cache4k}"
+  val cache4kJvm = ivy"io.github.reactivecircus.cache4k:cache4k-jvm:${versions.cache4k}"
+
+  // plugins
+  val kotlinxSerializationPlugin =
+    ivy"org.jetbrains.kotlin:kotlin-serialization-compiler-plugin:${versions.kotlin}"
+}
+
+object `package` extends RootModule {
+
+  // comment to disable test for the module
+  val modulesWithTestingEnabled = Set[CoursierModule](
+// these modules don't have hand-written tests
+//    `arrow-libs`.core.`arrow-annotations`,
+//    `arrow-libs`.core.`arrow-cache4k`,
+//    `arrow-libs`.core.`arrow-core-high-arity`,
+
+    `arrow-libs`.core.`arrow-atomic`,
+    `arrow-libs`.core.`arrow-autoclose`,
+    `arrow-libs`.core.`arrow-core-retrofit`,
+    `arrow-libs`.core.`arrow-core-serialization`,
+    `arrow-libs`.core.`arrow-core`,
+    `arrow-libs`.core.`arrow-eval`,
+    `arrow-libs`.core.`arrow-functions`,
+    `arrow-libs`.fx.`arrow-collectors`,
+    `arrow-libs`.fx.`arrow-fx-coroutines`,
+    `arrow-libs`.fx.`arrow-fx-stm`,
+// Cannot test this one - it relies on the module dependency in the .jar format, while Mill provides .class files
+//    `arrow-libs`.optics.`arrow-optics-ksp-plugin`,
+    `arrow-libs`.optics.`arrow-optics-reflect`,
+    `arrow-libs`.optics.`arrow-optics`,
+
+    `arrow-libs`.resilience.`arrow-resilience`
+  )
```

Comment:
After I better understand why that is needed at all, I'd like to move this into it's own support trait, so we can just mix in all the customized `compile` and `test` tasks, so the actual domain setup is free of these quirks.